### PR TITLE
Fix for collapsing lists - hide the label when list does not exceed max number

### LINKED ISSUE #101: Improved list for authors and institutions
Authors threshold: show first 9 and the last one
Institutions threshold: 3 lines

### PR DESCRIPTION
**Pull request by scottaubrey**
The label is currently visible regardless of list length.

Also:
- refactored the code a bit to use mixins rather than repeat the logic at each breakpoint. 
- Added more comments on the selector logic to remind me and others what we're selecting and why.

Fixes #101 101

### CODE REVIEW

**Comment by discodavey** [comment]
@scottaubrey was the alpha order fixing done through the linting?

**Comment by scottaubrey** [comment]
> @scottaubrey was the alpha order fixing done through the linting?

@discodavey I just noticed they weren't in alpha order and fixed it. I don't think it was flagged by the linter (as the commit prior passed)

**Comment by scottaubrey** [comment]
Preview environment: https://pr-325--epp.elifesciences.org/

**Comment by discodavey** [comment]
> > @scottaubrey was the alpha order fixing done through the linting?
> 
> @discodavey I just noticed they weren't in alpha order and fixed it. I don't think it was flagged by the linter (as the commit prior passed)

That's strange as it should have been flagged as I am sure we have put in a lint for alpha order

**Review by discodavey — COMMENTED** [review]
Do we actually need any CSS on the server side. I thought the idea of having the client side was to put the CSS there and then pull it through?

File: src/sass/_settings.scss [hunk 1]
```diff
@@ -70,7 +70,7 @@ $author-list-large-max: 10;
 $author-list-medium-max: 3;
 $author-list-small-max: 3;
 
-$institution-list-extra-large-max: 3;
-$institution-list-large-max: 3;
-$institution-list-medium-max: 2;
-$institution-list-small-max: 2;
+$affiliation-list-extra-large-max: 3;
```
Comment (discodavey) [inline src/sass/_settings.scss:73]: Any reason for this to change to affilation as on Journal these are known as institutions. Has the naming of these changed over the years?

Comment (scottaubrey) [reply src/sass/_settings.scss:73]: Spoke with Dave. This is Encoda's terminology. We could change all SASS to institutions, but encoda will still be giving us affiliations, and we still have to map the terms. We agreed to keep affiliations in server for the short term, and we'll map affiliation -> institutions on the client when the time comes. Hopefully, we will then also at some point remove encoda and be able to map all the way back to source institutions.

**Review by discodavey — APPROVED** [review]

**Pull request merged**